### PR TITLE
More safely delete files

### PR DESCRIPTION
**Pull request by stefan-kolb**
See #245

- confirmation before file deletion
- change language strings so the user knows this action is permanent

### CODE REVIEW

File: src/main/java/net/sf/jabref/gui/fieldeditors/FileListEditor.java [hunk 1]
```diff
@@ -218,13 +218,18 @@ public void actionPerformed(ActionEvent actionEvent) {
...
             int row = getSelectedRow();
+
             // no selection
-            if (row != -1) {
+            if (row == -1) {
+                return;
+            }
 
+            int userConfirm = JOptionPane.showConfirmDialog(null, Localization.lang("Are you sure you want to delete this file?"), Localization.lang("Warning"), JOptionPane.YES_NO_OPTION);
+            if (userConfirm == JOptionPane.YES_OPTION) {
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/gui/fieldeditors/FileListEditor.java]: Please don't use simple YES/NO Buttons: 
 http://www.uxdesignedge.com/2010/06/are-you-sure-how-to-write-effective-confirmations/

Better add explicit named Buttons Delete file permanently and Cancel:
As I just did this yesterday for another case, here is a sample code:

```
String[] options = {Localization.lang("Move file"), Localization.lang("Cancel")}; 
int dialogResult = JOptionPane.showOptionDialog(frame, "Move file to file directory" + fileDir.get(),
              "Move",
                   JOptionPane.INFORMATION_MESSAGE, JOptionPane.YES_NO_CANCEL_OPTION, null, options, options[0]);
            if (dialogResult == JOptionPane.YES_OPTION) {
```

Comment (stefan-kolb) [inline src/main/java/net/sf/jabref/gui/fieldeditors/FileListEditor.java]: Hm, one can argue if this really adds so much value. I think the functionality is clear with the question `Do you really want to delete this file?`. It just adds more Strings that need to be translated. However, I'm OK with implementing this. 

In general, what do @JabRef/developers think about this?

Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/gui/fieldeditors/FileListEditor.java]: There are just a few more Strings, but it really helps to improve usability a  lot! Here is a more detailed explanation . http://ux.stackexchange.com/questions/756/what-are-some-alternatives-to-the-phrase-are-you-sure-you-want-to-xyz-in-confi

See also https://github.com/koppor/jabref/issues/149

Comment (stefan-kolb) [inline src/main/java/net/sf/jabref/gui/fieldeditors/FileListEditor.java]: I understand it, tho not sure if I really agree 😄

**Pull request merged**